src/backend/distributed/commands/schema_based_sharding.c

```diff
@@ -8,28 +8,59 @@
 
 #include "postgres.h"
 #include "miscadmin.h"
+
+#include "access/genam.h"
+#include "catalog/catalog.h"
 #include "catalog/pg_namespace_d.h"
 #include "commands/extension.h"
 #include "distributed/argutils.h"
 #include "distributed/backend_data.h"
 #include "distributed/colocation_utils.h"
 #include "distributed/commands.h"
+#include "distributed/listutils.h"
 #include "distributed/metadata_sync.h"
+#include "distributed/metadata/distobject.h"
+#include "distributed/metadata_utility.h"
 #include "distributed/multi_partitioning_utils.h"
 #include "distributed/tenant_schema_metadata.h"
-#include "distributed/metadata/distobject.h"
+#include "distributed/worker_shard_visibility.h"
 #include "utils/builtins.h"
+#include "utils/fmgroids.h"
 #include "utils/lsyscache.h"
 #include "utils/syscache.h"
 
 
-PG_FUNCTION_INFO_V1(citus_internal_unregister_tenant_schema_globally);
-
+/*
+ * SchemaTables is used to classify different table types under a schema
+ */
+typedef struct SchemaTables
+{
+	Oid schemaId;
+	List *hashDistributedRelationIdList;
+	List *rangeDistributedRelationIdList;
+	List *appendDistributedRelationIdList;
+	List *referenceRelationIdList;
+	List *singleShardRelationIdList;
+	List *citusLocalRelationIdList;
+	List *localRelationIdList;
+} SchemaTables;
```

Comment:
A few nit comments regarding this struct:

* Do we really need schemaId?
* I'd place singleShardRelationIdList right after appendDistributedRelationIdList to keep the lists related to distributed tables together.
* Should we use "table" instead of "relation" when naming these elements, as we usually do?
* Renaming localRelationIdList to postgresTableIdList might make sense because, e.g., relation could mean a view in different contexts.